devlib/utils/ssh.py

```diff
@@ -308,12 +303,11 @@ def _scp(self, source, dest, timeout=30):
         # only specify -P for scp if the port is *not* the default.
         port_string = '-P {}'.format(self.port) if (self.port and self.port != 22) else ''
         keyfile_string = '-i {}'.format(self.keyfile) if self.keyfile else ''
-        command = '{} -r {} {} {} {}'.format(scp, keyfile_string, port_string, source, dest)
+        command = '{} -r {} {} {} {}'.format(scp, quote(keyfile_string), port_string, quote(source), quote(dest))
```

Comment:
I think so and I agree for passwords (unless we could detect something mandatory was missing early ourselves) unlike here where it is optional so it will depend on the situation as to the correct approach.